toolchain/check/context.h

```diff
@@ -387,6 +390,9 @@ class Context {
     // The name scope associated with this entry, if any.
     SemIR::NameScopeId scope_id;
 
+    // The previous state of name_lookup_has_load_error_, restored on pop.
+    bool prev_name_lookup_has_load_error;
```

Comment:
Why do we use a backup / restore approach here, rather than always looking at the state on the top of the scope stack when querying `name_lookup_has_load_error_`? (I'm not opposed to this if there's a good reason, but I think maintaining this flag alongside other current-scope information like the `names` set seems simpler if either would work.)

Reply:
I don't think that'd work:

```
import library "broken";

class C {
  fn Foo() {
    // Top of the scope stack has no load error, but this does name lookup into the package.
    var a: i32 = y;
  }

  // This pops the scope to `C`, `C`'s name scope does not have a load error, but
  // this still shouldn't print because it's unqualified name lookup.
  var b: i32 = y;
}
```

Reply:
The idea would be that when you push a new scope, its "name lookup has load error" flag is set to that of the enclosing scope.